imap_processing/cdf/utils.py

```diff
@@ -98,7 +105,21 @@ def write_cdf(
         + date_string
         + f"_v{data.attrs['Data_version']}.cdf"
     )
-    filename_and_path = os.path.join(directory, filename)
+
+    if directory is None:
```

Comment:
We could get rid of directory altogether and force everyone to go through the `imap_processing.DATA_DIR`? Then in all tests we'd just set that to a temporary location.

Reply:
This directory is to save processed data file and then I assume, we upload data from this directory to data bucket using upload API. Will we want to use `imap_processing.DATA_DIR`? If so, then yeah.

Reply:
will this go away now?

Reply:
See comment above. Leaving for now. We can remove it later if we want.